src/test/java/htsjdk/samtools/util/BlockCompressedInputStreamTest.java

```diff
@@ -86,4 +91,142 @@ public void available_should_return_number_of_bytes_left_in_current_block() thro
 			}
 		}
 	}
+
+    @Test
+    public void testCustomInflaterFileInput() throws Exception {
+        final File f = File.createTempFile("testCustomInflater.", ".gz");
+        f.deleteOnExit();
+        System.out.println("Creating file " + f);
+
+        final List<String> linesWritten = new ArrayList<>();
+        final BlockCompressedOutputStream bcos = new BlockCompressedOutputStream(f, 5);
+        String s = "Hi, Mom!\n";
+        bcos.write(s.getBytes()); //Call 1
+        linesWritten.add(s);
+        s = "Hi, Dad!\n";
+        bcos.write(s.getBytes()); //Call 2
+        linesWritten.add(s);
+        bcos.flush();
+        final StringBuilder sb = new StringBuilder(BlockCompressedStreamConstants.DEFAULT_UNCOMPRESSED_BLOCK_SIZE * 2);
+        s = "1234567890123456789012345678901234567890123456789012345678901234567890\n";
+        while (sb.length() <= BlockCompressedStreamConstants.DEFAULT_UNCOMPRESSED_BLOCK_SIZE) {
+            sb.append(s);
+            linesWritten.add(s);
+        }
+        bcos.write(sb.toString().getBytes()); //Call 3
+        bcos.close();
+
+        int[] inflateCalls = {0}; //Note: using and array is a HACK to fool the compiler
+        class MyInflater extends Inflater {
```

Comment:
Move `MyInflater` out of this method and into class scope, make it `private static`, refactor to allow the `inflateCalls` array to be passed in to the constructor and stored as a field, and share it between test cases instead of repeating the class definition in multiple methods.